File: internal/model/balance_log.go
package model

import "time"

type BalanceLog struct {
	ID           uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	TenantID     uint64    `gorm:"type:bigint;index;not null" json:"tenant_id"`
	KeyID        uint64    `gorm:"type:bigint;index;not null" json:"key_id"`
	KeyAlias     string    `gorm:"type:varchar(255);not null" json:"key_alias"`
	Delta        int64     `gorm:"type:bigint;not null" json:"delta"`
	BeforeAmount int64     `gorm:"type:bigint;not null" json:"before_amount"`
	AfterAmount  int64     `gorm:"type:bigint;not null" json:"after_amount"`
	Operator     string    `gorm:"type:varchar(100);not null" json:"operator"`
	Remark       string    `gorm:"type:varchar(500);not null;default:''" json:"remark"`
	CreatedAt    time.Time `gorm:"autoCreateTime;index" json:"created_at"`
}

func (BalanceLog) TableName() string { return "balance_logs" }

File: sdk/java/src/main/java/com/github/ezzzi_y/service/model/BalanceLog.java
package com.github.ezzzi_y.service.model;

import java.time.OffsetDateTime;

/**
 * 余额变更日志。
 */
public class BalanceLog {

    private final long id;
    private final long keyId;
    private final String keyAlias;
    private final long delta;
    private final long beforeAmount;
    private final long afterAmount;
    private final String operator;
    private final String remark;
    private final OffsetDateTime createdAt;

    public BalanceLog(long id, long keyId, String keyAlias, long delta,
                      long beforeAmount, long afterAmount,
                      String operator, String remark, OffsetDateTime createdAt) {
        this.id = id;
        this.keyId = keyId;
        this.keyAlias = keyAlias;
        this.delta = delta;
        this.beforeAmount = beforeAmount;
        this.afterAmount = afterAmount;
        this.operator = operator;
        this.remark = remark;
        this.createdAt = createdAt;
    }

    public long getId() {
        return id;
    }

    public long getKeyId() {
        return keyId;
    }

    public String getKeyAlias() {
        return keyAlias;
    }

    public long getDelta() {
        return delta;
    }

    public long getBeforeAmount() {
        return beforeAmount;
    }

    public long getAfterAmount() {
        return afterAmount;
    }

    public String getOperator() {
        return operator;
    }

    public String getRemark() {
        return remark;
    }

    public OffsetDateTime getCreatedAt() {
        return createdAt;
    }

    @Override
    public String toString() {
        return "BalanceLog{id=" + id + ", keyAlias='" + keyAlias
                + "', delta=" + delta + ", after=" + afterAmount + "}";
    }
}
